parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -226,6 +226,9 @@ private CompressionCodecName codec() {
       set("parquet.avro.write-old-list-structure", "false");
       MessageType type = ParquetSchemaUtil.convert(schema, name);
 
+      // Check that our metrics make sense
+      metricsConfig.validateProperties(schema);
```

Comment:
Is this the right place to do the validation?

If a user adds a bad property or performs some schema update that causes a validation error, that would break all writes to the table. To me, it doesn't seem like we are catching the problem early enough and possibly allowing a typo to break scheduled jobs.

What do you think about adding this validation when altering the table? `UpdateProperties` could check whether any properties starting with `write.metadata.metrics` were modified and run this. Similarly, `UpdateSchema` could run this as well, although I think that we should probably modify `UpdateSchema` to simply update the properties for column renames (if that's easily done).

Reply:
Makes sense, I'll move this over to UpdateProperties/UpdateSchema tomorrow :)

Reply:
I'd also cover table creation. I feel that we should fail table creation if the metrics config refers to non-existing columns. I have seen so many cases when people clone an example but forget to update the metrics config and they realize the problem only after ingestion is done.

Reply:
That's a good point. Table creation is a good time to validate this as well.

Reply:
I've put this in for table creation & schema + property evolution. Let me know if I hooked in the correct places.